include/secp256k1_bulletproofs.h

```diff
@@ -9,7 +9,62 @@ extern "C" {
 
 #include <stdint.h>
 
-/* TODO */
+/** Opaque structure representing a large number of NUMS generators */
+typedef struct secp256k1_bulletproofs_generators secp256k1_bulletproofs_generators;
+
+/** Allocates and initializes a list of NUMS generators.
+ *  Returns a list of generators, or NULL if allocation failed.
```

Comment:
This should probably rather say that it calls the error callback when allocation failed. And since the error callback is not supposed to return, documenting the return value here is not necessary. (The function should still return NULL if the error callback returns unexpectedly.) Same below.